packages/connect/src/device/Device.ts

```diff
@@ -334,12 +343,88 @@ export class Device extends TypedEmitter<DeviceEvents> {
         return this.acquirePromise;
     }
 
-    trezorPushNotificationHandler(message: number[]) {
+    private trezorPushNotificationHandler(message: number[]) {
         // TODO: now we are just emitting event, other logic will be implemented for some notifications.
         const decoded: DecodedTrezorPushNotification = tpn.decode(message);
         this.lifecycle.emit(DEVICE.TREZOR_PUSH_NOTIFICATION, decoded);
     }
 
+    private resetBootMode(mode: TrezorPushNotificationMode) {
+        _log.info(`mode changed to ${mode} reset Features and ThpState`);
+        // @ts-expect-error
+        this._features = undefined;
+        if (mode === TrezorPushNotificationMode.BootloaderMode) {
+            this.busy = 'bootloader-locked';
+            this._protocol = protocolV1;
+            this.thp?.resetState();
+            this.thp = undefined;
+        } else {
+            this.busy = 'rebooting';
+        }
+    }
+
+    private async setupBootMode(mode: TrezorPushNotificationMode) {
+        if (mode === TrezorPushNotificationMode.BootloaderMode && !this.features) {
+            _log.info('loading Features after unlock');
+            this.busy = 'rebooting';
+            await this.acquire();
+            await this.getFeatures();
```

Comment:
this.getFeatures can throw so we definitely need some catch-ways handling. cc @marekrjpolak